src/BigIntFromString.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * @since 0.5.11
+ */
+import * as t from 'io-ts'
+import { either } from 'fp-ts/lib/Either'
+import { NonEmptyString } from './NonEmptyString'
+
+/**
+ * @since 0.5.11
+ */
+export interface BigIntFromStringC extends t.Type<bigint, string, unknown> {}
+
+/**
+ * @example
+ * import { BigIntFromString } from 'io-ts-types/lib/BigIntFromString'
+ * import { right } from 'fp-ts/lib/Either'
+ * import { PathReporter } from 'io-ts/lib/PathReporter'
+ *
+ * assert.deepStrictEqual(BigIntFromString.decode('1'), right(BigInt(1)))
+ * assert.deepStrictEqual(PathReporter.report(BigIntFromString.decode('1.1')), ['Invalid value "1.1" supplied to : BigIntFromString'])
+ * assert.deepStrictEqual(PathReporter.report(BigIntFromString.decode('')), ['Invalid value "" supplied to : BigIntFromString'])
+ *
+ * @since 0.5.11
+ */
+export const BigIntFromString: BigIntFromStringC = new t.Type<bigint, string, unknown>(
+  'BigIntFromString',
+  // tslint:disable-next-line
+  (u): u is bigint => typeof u === 'bigint',
+  (u, c) =>
+    either.chain(t.string.validate(u, c), s => {
+      if (!NonEmptyString.is(s.trim())) {
+        return t.failure(u, c)
+      }
```

Comment:
Don't `trim`. Codecs should not make as few assumptions as possible. Trimming could be added by another Codec when needed.

Reply:
The trim is necessary because `BigInt('') === BigInt(0)` -- we want to avoid that to keep parity with NumberFromString here https://github.com/gcanti/io-ts-types/blob/master/src/NumberFromString.ts#L27 -- but `BigInt(' ') === BigInt(0)` as well. Seems like a trap to decode `BigInt('')` into a left but `BigInt(' ')` into a right

Reply:
I see, after 10 years still learning new oddities in JS type conversion 😉

Reply:
haha you and me both!